crates/oxc_ast/src/ast/js.rs

```diff
@@ -1018,6 +1018,12 @@ pub enum ForStatementInit<'a> {
     Expression(Expression<'a>),
 }
 
+impl<'a> ForStatementInit<'a> {
+    pub fn is_lexical_declaration(&self) -> bool {
```

Comment:
A small comment unrelated to this PR: I recommend having some sort of documentation about certain things like "what is a lexical declaration?".

Reply:
I added the following, this is from the ecma spec, you'll see a lot of these in the codebase.
```
    /// LexicalDeclaration[In, Yield, Await] :
    ///   LetOrConst BindingList[?In, ?Yield, ?Await] ;

```